addons/point_of_sale/models/product.py

```diff
@@ -277,14 +277,24 @@ class ProductTemplateAttributeLine(models.Model):
     def _load_pos_data_fields(self, config_id):
         return ['display_name', 'attribute_id', 'product_template_value_ids']
 
+    @api.model
+    def _load_pos_data_domain(self, data):
+        loaded_product_tmpl_ids = list({p['product_tmpl_id'] for p in data['product.product']['data']})
```

Comment:
```suggestion
        loaded_product_tmpl_ids = (p['product_tmpl_id'] for p in data['product.product']['data'])
```
doesn't this work? Why use a dictionary?

Reply:
No, it won't work. I used a set comprehension, and it's not a dictionary. I wanted to remove the duplicates.

Reply:
@vlst-odoo I forgot to ping you.

Reply:
at this point i'm more curious than anything :) Why would you bother with removing the duplicates? Do you think that removing the duplicates saves more time than sending the whole list to the orm ?

Reply:
@vlst-odoo I tested this, and the ORM does indeed include duplicate IDs in the query. After researching, I found that it's generally better to convert the list of IDs into a set before performing the query. This avoids unnecessary database queries and optimizes performance, especially when dealing with larger datasets.

Reply:
i see. It's quite funny that record`sets` contain duplicates 
![image](https://github.com/user-attachments/assets/7fb93b99-a535-4255-bea0-fd9ef0122a49)
I didn't know this.
Let's go with your solution then. Thanks for sharing this info.